tool/simulation/change_parameters.py

```diff
@@ -0,0 +1,264 @@
+2. First we iterate the files then the parameters of each files. At the end of iteration, we called
+   `execution()`.
+"""
+#====================================================================================================
+# Import packages
+#====================================================================================================
+import numpy as np
+import argparse
+import re
+import os
+import sys
+import subprocess
+
+
+
+#====================================================================================================
+# Global variables
+#====================================================================================================
+RECURSION_LIMIT = 1000  # Reset the recursion limit. (default in python is 1000)
+RETURN_FAIL    = False
+RETURN_SUCCESS = True
+
+
+
+#====================================================================================================
+# Classes
+#====================================================================================================
+class File():
+    def __init__( self, file_name, consts, paras, flag ):
+        """
+        file_name : string. The file name.
+        consts    : dict. The const parameter want to be specified.
+        paras     : dict with list element. The parameter want to be chenged.
+        flag      : bool. The flag file or not.
+        """
+        self.name   = file_name
+        self.paras  = paras
+        self.consts = consts
+        self.flag   = flag
+
+
+#====================================================================================================
+# Functions
+#====================================================================================================
+def iter_files( files, record_changed={}, **kwargs ):
+    # copy_files = files.copy() # work in python3 only
+    copy_files = list(files)
+
+    if copy_files == []:        # End of changing files
+        return execution( record=record_changed, **kwargs )
+
+    replace_file_class = copy_files.pop(0)
+    replace_file  = replace_file_class.name
+    replace_paras = replace_file_class.paras
+    replace_flag  = replace_file_class.flag
+    return iter_paras( replace_file, replace_paras, record_changed, replace_flag, rest_files=copy_files, **kwargs )
+
+def iter_paras( file_name, paras, record, flag, **kwargs ):
+    copy_paras  = paras.copy()
+    copy_kwargs = kwargs.copy()
+    copy_kwargs.pop("rest_files")
+    if copy_paras == {}:        # End of the changes
+        return iter_files( kwargs["rest_files"], **copy_kwargs )
+
+    replace_key = next(iter(copy_paras)) # Take the first key to replace
+    vals = copy_paras.pop(replace_key)
+    for val in vals:
+        if not kwargs["quite"]: print("File <%s> changing %s --> %s"%(file_name, replace_key, str(val)))
+        replace_parameter( file_name, replace_key, val, flag )
+        record[replace_key]=val
+        iter_paras( file_name, copy_paras, record, flag, **kwargs )  # repalce the next parameter
```

Comment:
- Capitalize the first letter (i.e., "r" --> "R") to be consistent with other comments. Check the entire script.
- Typo: "repalce"